frontend-react/src/components/ManagePublicKey/ManagePublicKey.tsx

```diff
@@ -1,18 +1,148 @@
-import React from "react";
-import { GridContainer } from "@trussworks/react-uswds";
+import React, { useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { GridContainer, Icon, SiteAlert } from "@trussworks/react-uswds";
 
 import { AuthElement } from "../AuthElement";
 import { withCatchAndSuspense } from "../RSErrorBoundary";
+import { USLink } from "../USLink";
+import Spinner from "../Spinner";
+import { showError } from "../AlertNotifications";
 import { MemberType } from "../../hooks/UseOktaMemberships";
+import {
+    validateFileType,
+    ManagePublicKeyAction,
+    CONTENT_TYPE,
+    FORMAT,
+    validateFileSize,
+} from "../../hooks/network/ManagePublicKey/ManagePublicKeyHooks";
+
+import { ManagePublicKeyChooseSender } from "./ManagePublicKeyChooseSender";
+import { ManagePublicKeyUpload } from "./ManagePublicKeyUpload";
+
+const LightbulbIcon = Icon.Lightbulb;
+
+const ManagePublicKeySwitchDisplay = () => {
+    const navigate = useNavigate();
+    const [action, setAction] = useState(ManagePublicKeyAction.SELECT_SENDER);
```

Comment:
I think it's fine to differentiate between steps with explicit step types like this, but I don't think we should be setting them directly in state.  Why not just infer that from whatever state we have stored already?